src/esp/gfx/CubeMapCamera.h

```diff
@@ -0,0 +1,125 @@
+// Copyright (c) Facebook, Inc. and its affiliates.
+// This source code is licensed under the MIT license found in the
+// LICENSE file in the root directory of this source tree.
+
+#ifndef ESP_GFX_CUBEMAPCAMERA_H_
+#define ESP_GFX_CUBEMAPCAMERA_H_
+
+#include <Magnum/GL/CubeMapTexture.h>
+#include <Magnum/Magnum.h>
+#include <Magnum/Math/Tags.h>
+#include "esp/core/esp.h"
+#include "esp/gfx/RenderCamera.h"
+
+namespace esp {
+namespace gfx {
+class CubeMapCamera : public RenderCamera {
+ public:
+  /**
+   * @brief Constructor
+   * @param node, the scene node to which the camera is attached
+   */
+  explicit CubeMapCamera(scene::SceneNode& node);
+  /**
+   * @brief Constructor
+   * @param node, the scene node to which the camera is attached
+   * @param eye, the eye position (parent node space)
+   * @param target, the target position (parent node space)
+   * @param up, the up direction (parent node space)
+   */
+  explicit CubeMapCamera(scene::SceneNode& node,
+                         const vec3f& eye,
+                         const vec3f& target,
+                         const vec3f& up);
+  /**
+   * @brief Constructor
+   * @param node, the scene node to which the camera is attached
+   * @param eye, the eye position (parent node space)
+   * @param target, the target position (parent node space)
+   * @param up, the up direction (parent node space)
+   */
+  explicit CubeMapCamera(scene::SceneNode& node,
+                         const Magnum::Vector3& eye,
+                         const Magnum::Vector3& target,
+                         const Magnum::Vector3& up);
+  virtual ~CubeMapCamera(){};
+  /**
+   * @brief Move the camera towards a specified cube face
+   * @param cubeSide, the cube map coordinate, see the following pictures
+   * NOTE: +Y is top
+   *           +----+
+   *           | -Y |
+   * +----+----+----+----+
+   * | -Z | -X | +Z | +X |
+   * +----+----+----+----+
+   *           | +Y |
+   *           +----+
+   * CAREFUL! the local transformation of the camera node will be set after
+   * calling this function.
+   * @return Reference to self (for method chaining)
+   */
+  CubeMapCamera& switchToFace(Magnum::GL::CubeMapCoordinate cubeSide);
+  /**
+   * @brief Overload, move the camera towards a specified cube face
+   * @param cubSideIndex, the index of the cube map coordinate.
+   * 0: +X
+   * 1: -X
+   * 2: +Y
+   * 3: -Y
+   * 4: +Z
+   * 5: -Z
+   *           +----+
+   *           | -Y |
+   * +----+----+----+----+
+   * | -Z | -X | +Z | +X |
+   * +----+----+----+----+
+   *           | +Y |
+   *           +----+
+   * @return Reference to self (for method chaining)
```

Comment:
The ASCII art will render pretty bad in the docs. Put it outside the `@param` documentation and wrap in a ``` block.

Reply:
I see. Good to know. Will do.